### PR TITLE
Properly escape fields passed to defrecord

### PR DESCRIPTION
**Pull request by whatyouhide**
Should close https://github.com/elixir-lang/elixir/issues/6060.

### CODE REVIEW

File: lib/elixir/test/elixir/record_test.exs [hunk 1]
```diff
@@ -137,7 +137,9 @@ defmodule RecordTest do
     charlist: 'abc'
 
   test "records with literal defaults" do
-    assert defaults(defaults()) == [
+    defaults = defaults()
```
Comment (josevalim) [inline lib/elixir/test/elixir/record_test.exs]: Maybe a comment here saying that we need to have this on its own line to detect any escaping issues which would be by passed with we just passed it to the macro below?

Comment (josevalim) [inline lib/elixir/test/elixir/record_test.exs]: Or maybe have two tests. One for `defaults/0` and another for `defaults/1`.

Comment (whatyouhide) [inline lib/elixir/test/elixir/record_test.exs]: Both suggestions make sense 👍

**Comment by whatyouhide** [comment]
Backported to 1.4.

**Pull request merged**